nexus/sdk-go.md

```diff
@@ -0,0 +1,352 @@
+// the name).
+opStartTransactionAlt4 := temporalnexus.NewWorkflowRunOperationWithOptions(
+	temporalnexus.WorkflowRunOperationOptions[MyInput, MyOutput]{
+		Name: "start-transaction",
+		Handler: func(ctx context.Context, input MyInput, opts nexus.StartOperationOptions) (temporalnexus.WorkflowHandle[MyOutput], error) {
+			// Run any arbitrary workflow.
+			return temporalnexus.ExecuteUntypedWorkflow[MyOutput](ctx, client.StartWorkflowOptions{
+				ID: input.ID,
+			}, "SomeOtherWorkflow", input1, input2, input3)
+		},
+	})
+```
+
+### Register an Operation with a Worker
+
+We define a `NexusOperationRegistry` interface with a single method `RegisterNexusOperation` and embed it in the
+`Registry` interface:
+
+```go
+import "github.com/nexus-rpc/sdk-go/nexus"
+
+Registry interface {
+	WorkflowRegistry
+	ActivityRegistry
+	NexusOperationRegistry // <-- New
+}
+
+NexusOperationRegistry interface {
+	// RegisterNexusOperation registers an operation with a worker. Panics if an operation with the same name has
+	// already been registered on this worker or if the worker has already been started. A worker will only poll for
+	// Nexus tasks if any operations are registered on it.
+	RegisterNexusOperation(op nexus.RegisterableOperation)
+}
+```
+
+**Usage**:
+
+```go
+myWorker.RegisterNexusOperation(myOperation)
+```
+
+## Starting an Operation from a Workflow
+
+The `ExecuteNexusOperation` API is modelled after `ExecuteChildWorkflow`, leveraging futures that can be used with the
+SDK's selectors.
+
+> NOTE: In the future, as the Go SDK adds support for typed futures, we will add a strongly typed variant of this API.
+
+```go
+package workflow
+
+// NexusOperationOptions are options for starting a Nexus Operation from a Workflow.
+type NexusOperationOptions struct {
+	ScheduleToCloseTimeout time.Duration
+}
+
+// NexusOperationExecution is the result of [NexusOperationFuture.GetNexusOperationExecution].
+type NexusOperationExecution struct {
+	OperationID string
+}
+
+// NexusOperationFuture represents the result of a Nexus Operation.
+type NexusOperationFuture interface {
+	Future
+	// GetNexusOperationExecution returns a future that is resolved when the operation reaches the STARTED state.
+	// For synchronous operations, this will be resolved at the same as the containing [NexusOperationFuture]. For
+	// asynchronous operations, this future is resolved independently.
+	// If the operation is unsuccessful, this future will contain the same error as the [NexusOperationFuture].
+	// Use this method to extract the Operation ID of an asynchronous operation. OperationID will be empty for
+	// synchronous operations.
+	GetNexusOperationExecution() Future
+}
+
+// ExecuteNexusOperation executes a Nexus Operation.
+func ExecuteNexusOperation(ctx Context, service string, operation string, input any, options NexusOperationOptions) NexusOperationFuture
```

Comment:
```suggestion
func ExecuteNexusOperation(ctx Context, service string, operation any, input any, options NexusOperationOptions) NexusOperationFuture
```

I should be able to provide an operation I defined elsewhere

Reply:
Good callout, thanks!

Reply:
Fixed.